yarn-project/aztec.js/src/account/contract/eip_1271_account_contract.ts

```diff
@@ -0,0 +1,35 @@
+import { Schnorr } from '@aztec/circuits.js/barretenberg';
+import { ContractAbi } from '@aztec/foundation/abi';
+import { CompleteAddress, NodeInfo, PrivateKey } from '@aztec/types';
+
+import Eip1271AccountContractAbi from '../../abis/schnorr_eip_1271_account_contract.json' assert { type: 'json' };
+import { Eip1271AccountEntrypoint } from '../entrypoint/eip_1271_account_entrypoint.js';
+import { AccountContract } from './index.js';
+
+/**
+ * Account contract that authenticates transactions using Schnorr signatures verified against
+ * the note encryption key, relying on a single private key for both encryption and authentication.
+ * Extended to pull verification data from the oracle instead of passed as arguments.
+ */
+export class Eip1271AccountContract implements AccountContract {
```

Comment:
I'm curious how you found this abstraction to implement your own account contract..? Was it easy, confusing..? Anything you'd change?